frontend/device/android/device.lua

```diff
@@ -254,12 +253,23 @@ function Device:performHapticFeedback(type)
 end
 
 function Device:retrieveNetworkInfo()
-    local ssid, ip, gw = android.getNetworkInfo()
-    if ip == "0" or gw == "0" then
+    local ok, type = android.getNetworkInfo()
+    ok, type = tonumber(ok), tonumber(type)
+    if not ok or not type or type == C.ANETWORK_NONE then
         return _("Not connected")
     else
-        local BD = require("ui/bidi")
-        return T(_("Connected to %1\n IP address: %2\n gateway: %3"), BD.wrap(ssid), BD.ltr(ip), BD.ltr(gw))
+        if type == C.ANETWORK_WIFI then
+            return _("Connected to Wi-Fi")
+        elseif type == C.ANETWORK_MOBILE then
+            return _("Connected to mobile data network")
+        elseif type == C.ANETWORK_ETHERNET then
+            return _("Connected to ethernet")
+        elseif type == C.ANETWORK_BLUETOOTH then
+            return _("Connected to bluetooth")
```

Comment:
As you know I'm in favor of wifi (and therefore possibly bluetooth by similar reasoning) but for now that should be Bluetooth in English.

```suggestion
            return _("Connected to Bluetooth")
```

Reply:
The same for Ethernet?